tests/BuildScriptGenerator.Tests/DotnetCore/GlobalJsonSdkResolverTest.cs

```diff
@@ -0,0 +1,1217 @@
+        }
+
+        [Fact]
+        public void LatestPatch_ReturnsLatestPatchVersion_IfMultiplePatchVersionsAreAvailable()
+        {
+            // Arrange
+            var expectedVersion = "3.1.103";
+            var availableVersions = new[] { "2.1.100", "3.1.101", "3.1.102", "3.1.103", "3.2.104" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.LatestPatch,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Theory]
+        [InlineData("true")]
+        [InlineData(null)]
+        public void LatestPatch_ReturnsLatestPreviewVersion_IfAllowPrereleaseIsTrue(string allowPrerelease)
+        {
+            // Arrange
+            var expectedVersion = "3.1.100-preview3-01445";
+            var availableVersions = new[]
+            {
+                "2.1.100",
+                "3.1.100-preview1-01445",
+                "3.1.100-preview2-01445",
+                "3.1.100-preview3-01445",
+            };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100-preview1-01445",
+                RollForward = GlobalJsonSdkResolver.LatestPatch,
+                AllowPreRelease = allowPrerelease,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void LatestPatch_ReturnsReleasedVersion_OverPreviewVersion_IfReleasedVersionIsAvailable_AndAllowPrereleaseIsTrue()
+        {
+            // Arrange
+            var expectedVersion = "3.1.100";
+            var availableVersions = new[]
+            {
+                "2.1.100",
+                "3.1.100-preview1-01445",
+                "3.1.100-preview2-01445",
+                "3.1.100-preview3-01445",
+                "3.1.100"
+            };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100-preview1-01445",
+                RollForward = GlobalJsonSdkResolver.LatestPatch,
+                AllowPreRelease = "true",
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void NoRollFoward_UseLatestPatchAsDefault()
```

Comment:
RollForward_ReturnsLatestPatchAsDefault_IfRollForwardIsNotPresent? Focusing so much on the naming lol..

Reply:
Nice! 

>  me focusing too much on the naming lol..

Actually this is good since it will help our team to have consistent and good names. Thanks!